### PR TITLE
Update relevant_feature_augmenter.py

### PR DESCRIPTION
**Pull request by yairst**
Fix bug in fit_transform() where filter_only_tsfresh_features is True and X is not empty. In this case, X_augmented does not include X.columns (inspect _fit_and_augment() output) resulting in pandas error: 'Passing list-likes to .loc or [] with any missing labels is no longer supported'. The changes in this commit fix this bug without hurting the other cases.
Also, change the description of _fit_and_augment() so that it matches the returned value, X_augmented

### CODE REVIEW

File: tests/units/transformers/test_relevant_feature_augmenter.py [hunk 1]
```diff
@@ -72,11 +76,14 @@ def test_evaluate_only_added_features_true(self):
         augmenter.fit(X, y)
         transformed_X = augmenter.transform(X.copy())
 
+        fit_transformed_X = augmenter.fit_transform(X, y)
+
         self.assertEqual(sum(["pre_feature" == column for column in transformed_X.columns]), 1)
+        self.assertEqual(sum(["pre_feature" == column for column in fit_transformed_X.columns]), 1)
 
-    def test_evaluate_only_added_features_false(self):
+    def filter_only_tsfresh_features_false(self):
```
Comment (nils-braun) [inline tests/units/transformers/test_relevant_feature_augmenter.py:84]: Is there a reason you removed the `test_` prefix from this function? This means it will not be picked up by most of the unittest frameworks

**Review by nils-braun — COMMENTED** [review]
Thank you very much for the fix! Looks reasonable to me except for the one small comment I have.

File: tests/units/transformers/test_relevant_feature_augmenter.py (hunk 1 above)
Comment (yairst) [reply tests/units/transformers/test_relevant_feature_augmenter.py:84]: deleted accidently. will be fixed
